src/ekat/ekat_physical_constants.hpp

```diff
@@ -0,0 +1,105 @@
+#ifndef EKAT_PHYSICAL_CONSTANTS
+#define EKAT_PHYSICAL_CONSTANTS
+
+#include "ekat/util/ekat_units.hpp"
+
+namespace ekat {
+
+template <typename RealType=double>
+class PhysicalConstant {
+  public:
+    enum Op {Times, Divide};
+
+    // no default
+    PhysicalConstant() = delete;
+
+    PhysicalConstant(const PhysicalConstant<RealType>&) = default;
+
+    // Constructor for single value (i.e., NIST std)
+    PhysicalConstant(const RealType& nvalue, const ekat::units::Units& units, const std::string& name) :
+      m_nist_val(nvalue), m_e3sm_val(nvalue), m_units(units), m_name(name), m_info("") {}
+
+    // Constructor for single value (i.e., NIST std)
+    PhysicalConstant(const RealType& nvalue, const ekat::units::Units& units, const std::string& name,
+      const std::string& info) :
+      m_nist_val(nvalue), m_e3sm_val(nvalue), m_units(units), m_name(name), m_info(info) {}
+
+
+    // Constructor for two different values (i.e., when E3SM value and NIST value are different)
+    PhysicalConstant(const RealType& evalue, const RealType& nvalue, const ekat::units::Units& units,
+      const std::string& name) : m_e3sm_val(evalue), m_nist_val(nvalue), m_units(units),
+      m_name(name), m_info("") {}
+
+    // Constructor for two different values (i.e., when E3SM value and NIST value are different)
+    PhysicalConstant(const RealType& evalue, const RealType& nvalue, const ekat::units::Units& units,
+      const std::string& name, const std::string& info) : m_e3sm_val(evalue), m_nist_val(nvalue), m_units(units),
+      m_name(name), m_info(info) {}
+
+
+    // Constructor for a constant that is defined by the product or quotient of 2 other constants.
+    PhysicalConstant(const PhysicalConstant<RealType>& lhs, const Op& op, const PhysicalConstant<RealType>& rhs,
```

Comment:
I think it would be easier if you overloaded op* and op/. Then one would simply do

```
auto gamma (cp/cv, "specific heat ratio");
```
which mean you need a copy ctor that takes another PhysicalConstant, with a different name. The overload of op* and op/ could simply use "name1 * name2" and "name1 / name2" for the name, and "info1; info2" for the info section.

Reply:
Thanks.  The only reason I did it this way was because of the `name` member variable.   I agree it's ugly.  

I'm happy to make this change, but I wonder for things like the universal gas constant, R = Avogadro*Boltzmann, is that name ok?  And if not, how do you change the name while also respecting `const`?

Reply:
Resolved this with a few thoughts:
1. units are all boiled-down to their fundamentals (i.e., Joules becomes kg m^2 / s^2) so it's consistent if constants record their fundamental origins too.
2. the in-source name can be different e.g., R_gas_universal, than the member variable name (again, similar to units like Newtons).
3. The overloaded operators are more intuitive and therefore less error prone (@jeff-cohere POLA).